docs/src/basics/structs.md

```diff
@@ -0,0 +1,37 @@
+# Immutable `struct` Handling
```

Comment:
Instead of being about structs, this page should be specifically about handling the problem types.

Reply:
So rename structs.md to problem.md?

Reply:
yeah and change the title.